firehose/firehose_kinesis.go

```diff
@@ -36,20 +40,31 @@ type FirehoseWriterConfig struct {
 	FlushSize int
 }
 
-func NewFirehoseWriter(config FirehoseWriterConfig, mockEndpoint string) (*FirehoseWriter, error) {
+// NewFirehoseWriter constructs a FirehoseWriter
+func NewFirehoseWriter(config FirehoseWriterConfig) (*FirehoseWriter, error) {
 	if config.FlushCount > 500 || config.FlushCount < 1 {
 		return nil, fmt.Errorf("FlushCount must be between 1 and 500 messages")
 	}
 	if config.FlushSize < 1 || config.FlushSize > 4*1024*1024 {
 		return nil, fmt.Errorf("FlushSize must be between 1 and 4*1024*1024 (4 Mb)")
 	}
-	return &FirehoseWriter{
-		conf:         &config,
-		mockEndpoint: mockEndpoint,
-		batchers:     map[string]batcher.Batcher{},
-	}, nil
+
+	sess := session.Must(session.NewSession(aws.NewConfig().WithRegion(config.Region).WithMaxRetries(4)))
+
+	f := &FirehoseWriter{
+		streamName:     config.StreamName,
+		firehoseClient: awsFirehose.New(sess),
```

Comment:
👍 I'll see if we can move back to that. The firehose package used it, I just thought it was mainly for mocking which wasn't yet pursued here so seemed like overkill.